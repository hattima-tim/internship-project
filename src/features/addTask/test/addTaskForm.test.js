import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { toMatchDiffSnapshot } from "snapshot-diff";
import AddTaskForm from "../addTaskForm";

expect.extend({ toMatchDiffSnapshot });

test("'add task' button is rendered", () => {
  const { container } = render(<AddTaskForm />);

  expect(container).toMatchSnapshot();
});

test("a form show up on 'add task' button click",async()=>{
    const user = userEvent.setup();
    const {asFragment} = render(<AddTaskForm/>);
    const firstRender = asFragment();
    const addTaskButton = screen.getByRole('button',{name:'Add Task'});

    await user.click(addTaskButton);
    
    expect(firstRender).toMatchDiffSnapshot(asFragment());
});
