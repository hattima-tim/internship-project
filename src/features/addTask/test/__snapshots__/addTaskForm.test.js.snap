// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`'add task' button is rendered 1`] = `
<div>
  <button
    class="absolute bottom-4 right-4 mr-2 inline-block rounded bg-indigo-600 px-8 py-3 text-sm font-medium text-white transition hover:scale-110 hover:shadow-xl focus:outline-none focus:ring active:bg-indigo-500"
  >
    Add Task
  </button>
</div>
`;
